### PR TITLE
util: Disable 3DES ciphers for TLS connections

### PR DESCRIPTION
**Pull request by dveeden**
### What problem does this PR solve?

Problem Summary:

Depending on the version of Go that is used, etc some insecure TLS ciphersuites are enabled.

This removes the following ciphersuites:
* TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA
* TLS_RSA_WITH_3DES_EDE_CBC_SHA

Note that https://docs.pingcap.com/tidb/stable/enable-tls-between-clients-and-servers didn't list these ciphersuites.

See also:
* https://csrc.nist.gov/news/2017/update-to-current-use-and-deprecation-of-tdea

### What is changed and how it works?

How it Works:

`tls.CipherSuites()` only lists ciphersuite that have `Insecure=false`, but in some cases this includes 3DES based ciphersuites, and these are explicitly removed from the list before using it in the `tls.Config`

### Check List 

Tests 

- [x] Manual test (add detailed scripts or steps below)

```
openssl s_client -starttls mysql -connect 127.0.0.1:4000 -tls1_2 -cipher ECDHE-RSA-DES-CBC3-SHA
```

Side effects

- [x] Breaking backward compatibility

This *intentionally* removes compatibility with 3DES ciphersuites. This is not expected to impact any users as this is not commonly used for MySQL connections.

### Release note 

```release-note
Support for 3DES based TLS ciphersuites was removed
```

### CODE REVIEW

File: util/misc.go [hunk 1]
```diff
@@ -513,12 +513,31 @@ func LoadTLSCertificates(ca, key, cert string, autoTLS bool, rsaKeySize int) (tl
...
 	}
+
+	// This excludes ciphers listed in tls.InsecureCipherSuites() and can be used to filter out more
+	var cipherSuites []uint16
+	var cipherNames []string
+	for _, sc := range tls.CipherSuites() {
+		switch cipherName := sc.Name; cipherName {
+		case "TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA":
+			fallthrough
+		case "TLS_RSA_WITH_3DES_EDE_CBC_SHA":
```
Comment (kennytm) [inline util/misc.go:524]:
```suggestion
		case "TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA", "TLS_RSA_WITH_3DES_EDE_CBC_SHA":
```

File: util/misc.go [hunk 2]
```diff
@@ -513,12 +513,29 @@ func LoadTLSCertificates(ca, key, cert string, autoTLS bool, rsaKeySize int) (tl
...
+
+	// This excludes ciphers listed in tls.InsecureCipherSuites() and can be used to filter out more
+	var cipherSuites []uint16
+	var cipherNames []string
+	for _, sc := range tls.CipherSuites() {
+		switch cipherName := sc.Name; cipherName {
+		case "TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA", "TLS_RSA_WITH_3DES_EDE_CBC_SHA":
+			logutil.BgLogger().Info("Disabling weak cipherSuite", zap.String("cipherSuite", cipherName))
+		default:
+			cipherNames = append(cipherNames, sc.Name)
```
Comment (kennytm) [inline util/misc.go:525]: use `cipherName` here? or just don't introduce `cipherName := sc.Name` on line 521?

**Review by kennytm — APPROVED** [review]
LGTM, but some nitpick on variable use.

File: util/misc.go (hunk 2 above)
Comment (dveeden) [reply util/misc.go:525]: refactored this a bit, PTAL

**Review by morgo — APPROVED** [review]

**Pull request merged**